docs/ui-components/line-nos.md

```diff
@@ -40,90 +44,62 @@ Some code
 def foo
   puts 'foo'
 end
-{% endhighlight %}
-{% endcapture %}
-{% include fix_linenos.html code=code %}
-{% assign code = nil %}
+{% endhighlight %}{% endraw %}
+```
 
-{: .warning }
-The following generates **incorrect** and **invalid** HTML. It should not be used as a positive example; the improper layout (with the broken HTML tags) is intentional.
+If this is directly placed within a file processed by Jekyll (via Just the Docs), the following markup will be generated:
 
-❌ With the compression options used for the theme docs, the following example illustrates
-the **incorrect** formatting arising from the incompatibility of HTML compression
-and the non-conforming HTML produced by Jekyll for line numbers:
+```html
+<figure class="highlight">><code class="language-ruby" data-lang="ruby"><div class="table-wrapper"><table class="rouge-table"><tbody><tr><td class="gutter gl"><pre class="lineno">1
+2
+3
+</pre><td class="code"><pre><span class="k">def</span> <span class="nf">foo</span>
+  <span class="nb">puts</span> <span class="s1">'foo'</span>
+<span class="k">end</span>
+</pre></figure>
+```
 
-{% highlight ruby linenos %}
-def foo
-  puts 'foo'
-end
-{% endhighlight %}
+This HTML is invalid; in particular, there are two issues:
+
+1. there are many missing closing tags, which produces visually garbled output
+2. a `<table>` is placed within a `<code>` element, which is syntactically invalid HTML (but is often allowed by browsers)
+
+To emphasize this first difference, here is the same markup output, indented by tag:
+
+```html
+<figure class="highlight">
+  >
+  <code class="language-ruby" data-lang="ruby">
+    <div class="table-wrapper">
+      <table class="rouge-table">
+        <tbody>
+          <tr>
+            <td class="gutter gl">
+              <pre class="lineno">
+1
+2
+3
+              </pre>
+              <td class="code">
+                <pre>
+<span class="k">def</span>
+<span class="nf">foo</span>
+<span class="nb">puts</span>
+<span class="s1">'foo'</span>
+<span class="k">end</span>
+                </pre>
+</figure>
+```
+
+In order, there are missing `</td>`, `</td>`, `</tr>`, `</tbody>`, `</table>`, `</div>`, and `</code>` tags. As a result, the following elements of the page - including the site footer - become visually garbled as browsers attempt to recover gracefully.
+
+Prior workarounds we suggested (such as [Dmitry Hrabrov's in `jekyll-compress-html`#71](https://github.com/penibelst/jekyll-compress-html/issues/71#issuecomment-188144901)) resolve the missing tag problem. However, they still place a `<table>` within a `<code>` element. The HTML spec normatively outlines that `<code>` elements should only contain "[phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2)", of which `<table>` is not a part of ([spec ref](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-code-element)). Thus, we no longer officially recommend these workarounds.
```

Comment:
```suggestion
Prior workarounds we suggested (such as [Dmitry Hrabrov's in `jekyll-compress-html`#71](https://github.com/penibelst/jekyll-compress-html/issues/71#issuecomment-188144901)) resolve the missing tag problem. However, they still place a `<table>` within a `<code>` element. The HTML spec normatively states that `<code>` elements should only contain "[phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2)", which does not include `<table>` ([spec ref](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-code-element)). To avoid Invalid HTML, the previously-suggested workaround using the current version of `_includes/fix_linenos.html` should _not_ be used! 
```